src/data/list/basic.lean

```diff
@@ -590,6 +593,10 @@ reverse_involutive.injective
 @[simp] theorem reverse_inj {l₁ l₂ : list α} : reverse l₁ = reverse l₂ ↔ l₁ = l₂ :=
 reverse_injective.eq_iff
 
+lemma reverse_eq_iff {l l' : list α} :
+  l.reverse = l' ↔ l = l'.reverse :=
+by rw [←reverse_reverse l', reverse_inj, reverse_reverse]
```

Comment:
This is true for all maps satisfying `function.involutive` (like the group inverse function), but I can't find that in mathlib. It would be nice (though not required) if you can prove this for all involutive functions and derive `neg_eq_iff_neg_eq` and similar from it.